### PR TITLE
CROW routes and Deserialization suupport (from_json impls)

### PR DESCRIPTION
**Pull request by davepl**
This adds the required serialization support for the major types as well as CROW routes to CRxD the items in the server collection.

I've added to_json and from_json at global scope so that nlohmann can "silently" do type conversions via json.

For effects, each effect now implements interface-defined methods for ToJson() and FromJson(), and the effect factory will use those to serialize and deserialize live effects.  

This also takes palettes back to vectors, and unifies the StandardPalettes into one place.

It adds SetPixelF, the fractional pixel draw for smooth scrolling, etc.  I moved RandomInt and RandomDouble to utilities, nuked the old green test effect, added the bar graphs, etc.

### CODE REVIEW

File: canvas.h [hunk 1]
```diff
@@ -95,3 +93,26 @@ class Canvas : public ICanvas
     uint32_t _height;
     vector<unique_ptr<ILEDFeature>> _features;
 };
+
+
+#include "ledfeature.h"
```
Comment (davepl) [inline canvas.h:98]: I have an include order dependency that means LEDFeature has to include Canvas and Canvas needs LEDFeature.  Having this include here is a dirty hack until I figure out a better way to address it.... hopefully

**Comment by davepl** [comment]
One more thing!  I’ve started adding spdlog logging.  On the Mac, that’s as simple as “brew install spdlog” and then I added the include folder to the makefile.

But when I check in, it won’t work in CI as CI doesn’t have spdlog.  

How do you want to coordinate that?

Thanks!
Dave

**Comment by rbergen** [comment]
Yeah, that requires an update to .github/workflows/CI.yml. I've taken the
liberty to merge a micro-PR to add spdlog to "main", and also push that
change to "controller" on PSL/NDSCPP.

You'll obviously have to git pull that change locally.

Cheers,
Rutger

**Comment by davepl** [comment]
Thanks much!

**Comment by rbergen** [comment]
...and the ledmon Makefile also needed -lfmt to be added to LIBS. I pushed
that update to "controller" too.

You're welcome!

**Pull request merged**